src/Common/src/TypeSystem/CodeGen/TargetDetails.CodeGen.cs

```diff
@@ -0,0 +1,31 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using Debug = System.Diagnostics.Debug;
+
+namespace Internal.TypeSystem
+{
+    // Extension to TargetDetails related to code generation
+    partial class TargetDetails
+    {
+        public TargetDetails(TargetArchitecture architecture, TargetOS targetOS, TargetAbi abi, MaximumSimdVectorLength simdVectorLength)
+            : this(architecture, targetOS, abi)
+        {
+            MaximumSimdVectorLength = simdVectorLength;
+        }
+
+        public MaximumSimdVectorLength MaximumSimdVectorLength
+        {
+            get;
```

Comment:
Now that I had a chance to think about this a little on the way to get lunch: would we just want to introduce this as a `TargetArchitecture` flavor? E.g. `TargetArchitecture.x64`, `TargetArchitecture.x64_Avx`, with a `IsX64` convenience property on `TargetDetails`? (Similar for other arches, in the future.) Yes, it feels weird to put instruction set details on a type system object, but as we see here, this does affect type layout, and it is a type system concern.